### PR TITLE
Include contract packages in the data directory when building the release

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR adds functionality to include contract packages as part of the autonomy PYPI release and fixes the `get_contract` method

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: autonomy/__init__.py [hunk 1]
```diff
@@ -19,4 +19,9 @@
 
 """A framework for the creation of autonomous agent services."""
 
+from pathlib import Path
+
 from .__version__ import __version__
+
+
+AUTONOMY_DIR = Path(__file__).resolve().parent
```
Comment (angrybayblade) [inline autonomy/__init__.py:27]: similar to `aea.AEA_DIR`

File: Makefile [hunk 2]
```diff
@@ -142,8 +142,16 @@ test-all:
 install: clean
 	python3 setup.py install
 
+.PHONY: eject-contracts
+eject-contracts:
+	@for contract in component_registry registries_manager service_registry ; do \
+		echo Updating $$contract contract; \
+    	rm -rf autonomy/data/contracts/$$contract ; \
+		cp -r packages/valory/contracts/$$contract autonomy/data/contracts/$$contract ; \
+	done
```
Comment (angrybayblade) [inline Makefile:151]: Include contract packages in the `data/` directory instead of installing them as a separate package

File: Makefile [hunk 3]
```diff
@@ -142,8 +142,16 @@ test-all:
...
+
 .PHONY: dist
-dist: clean
+dist: clean eject-contracts
```
Comment (angrybayblade) [inline Makefile:154]: Ejecting contracts before the PYPI release build

File: autonomy/chain/mint.py [hunk 4]
```diff
@@ -118,10 +117,19 @@ def verify_and_fetch_token_id_from_event(
 
 def get_contract(public_id: PublicId) -> Contract:
     """Load contract for given public id."""
-    contract_dir = RELATIVE_CONTRACTS_DIR / public_id.name
+
+    # check if a local package is available
+    contract_dir = CONTRACTS_DIR_LOCAL / public_id.name
     if contract_dir.exists():
-        return contract_dir
-    return ROOT_DIR / contract_dir
```
Comment (angrybayblade) [inline autonomy/chain/mint.py:124]: This method should return an instance of a contract not a path to the contract. idk how the mypy missed this 🤔

File: setup.py [hunk 5]
```diff
@@ -96,12 +96,15 @@ def parse_readme():
...
+                "data/contracts/registries_manager/*",
+                "data/contracts/registries_manager/build/*",
+                "data/contracts/service_registry/*",
+                "data/contracts/service_registry/build/*",
                 "test_tools/data/*",
             ],
-            "packages": [
-                "valory/contracts/component_registry/*",
-                "valory/contracts/registries_manager/*",
-            ],
```
Comment (angrybayblade) [inline setup.py:104]: Same as above, not shipping the packages as separate package instead include them in the data dir

**Review by solarw — APPROVED** [review]
LGTM
can not say it's the most elegant solution cause file copying. but it does the trick.
looks like there is no another solution

**Comment by angrybayblade** [comment]
Failures (on both ubuntu and window)

```
FAILED tests/test_autonomy/test_cli/test_analyse/test_check_handlers.py::TestCheckHandlers::test_check_handlers_fail
FAILED tests/test_autonomy/test_cli/test_analyse/test_check_handlers.py::TestCheckHandlers::test_check_handlers
```

Looks flaky

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**